iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -284,4 +333,40 @@ private static String getCatalogType(Configuration conf, String catalogName) {
       }
     }
   }
+
+  public static class TableCache {
+    private static final Cache<String, Table> tableCache = Caffeine.newBuilder()
+        .expireAfterAccess(12, TimeUnit.HOURS).build();
+    private static final Set<String> keys = new TreeSet<>();
```

Comment:
Instead of maintaining the key set manually, you might be able to get the keys in the hook method by using `tableCache.getNativeCache().asMap().keySet()`?

Reply:
Right, I don't know how I missed that. I was looking for a map representation of the cache entries.